aptible/resource_metric_drain.go

```diff
@@ -0,0 +1,209 @@
+package aptible
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/aptible/go-deploy/aptible"
+	"github.com/go-openapi/strfmt"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceMetricDrain() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceMetricDrainCreate,
+		ReadContext:   resourceMetricDrainRead,
+		DeleteContext: resourceMetricDrainDelete,
+		CustomizeDiff: resourceMetricDrainValidate,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceMetricDrainImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"metric_drain_id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"env_id": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"handle": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"drain_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"influxdb_database", "influxdb", "datadog"}, false),
+			},
+			"database_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"url": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validateURL,
+			},
+			"username": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"password": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				ForceNew:  true,
+				Sensitive: true,
+			},
+			"database": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"api_key": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				ForceNew:  true,
+				Sensitive: true,
+			},
+			"series_url": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validateURL,
+			},
+		},
+	}
+}
+
+var resourceMetricDrainAttrs = map[string]ResourceAttrs{
+	"influxdb_database": {
+		Required:   []string{"database_id"},
+		NotAllowed: []string{"url", "username", "password", "database", "api_key", "series_url"},
+	},
+	"influxdb": {
+		Required:   []string{"url", "username", "password", "database"},
+		NotAllowed: []string{"database_id"},
+	},
+	"datadog": {
+		Required:   []string{"api_key"},
+		NotAllowed: []string{"database_id"},
+	},
+}
+
+func resourceMetricDrainValidate(_ context.Context, diff *schema.ResourceDiff, _ interface{}) error {
+	d := ResourceDiff{ResourceDiff: diff}
+	drainType := d.Get("drain_type").(string)
+	var err error
+
+	allowedAttrs := resourceMetricDrainAttrs[drainType]
```

Comment:
might want an error if not found if that's even possible

Reply:
Done